src/main.rs

```diff
@@ -88,20 +90,22 @@ impl Deno {
         Deno { ptr: ptr }
     }
 
-    fn execute(&self, js_filename: &str, js_source: &str) -> bool {
+    fn execute(&mut self, js_filename: &str, js_source: &str) -> Result<(), DenoException> {
         let filename = CString::new(js_filename).unwrap();
         let source = CString::new(js_source).unwrap();
         let r = unsafe { deno_execute(self.ptr, filename.as_ptr(), source.as_ptr()) };
-        r != 0
-    }
-
-    fn last_exception(&self) -> &str {
-        let ptr = unsafe { deno_last_exception(self.ptr) };
-        let cstr = unsafe { CStr::from_ptr(ptr) };
-        cstr.to_str().unwrap()
+        if r != 0 {
```

Comment:
Is it possible to remove the additional block?
```rust
if r != 0 {
  Ok(())
}
let ptr = unsafe { deno_last_exception(self.ptr) };
let cstr = unsafe { CStr::from_ptr(ptr) };
Err(cstr.to_str().unwrap())
```

Reply:
Still needs to branch on the result of `deno_execute`.
Maybe something like:
```
    fn execute(&mut self, js_filename: &str, js_source: &str) -> Result<(), DenoException> {
        let filename = CString::new(js_filename).unwrap();
        let source = CString::new(js_source).unwrap();
        let r = unsafe { deno_execute(self.ptr, filename.as_ptr(), source.as_ptr()) };
        if r == 0 {
            let ptr = unsafe { deno_last_exception(self.ptr) };
            let cstr = unsafe { CStr::from_ptr(ptr) };
            Err(cstr.to_str().unwrap())
        }
        Ok(())
    }
```

Reply:
👌